std/experimental/allocator/building_blocks/affix_allocator.d

```diff
@@ -60,36 +63,39 @@ struct AffixAllocator(Allocator, Prefix, Suffix = void)
         enum uint alignment = Prefix.alignof;
     }
 
-    /**
-    If the parent allocator $(D Allocator) is stateful, an instance of it is
-    stored as a member. Otherwise, $(D AffixAllocator) uses
-    `Allocator.instance`. In either case, the name $(D _parent) is uniformly
-    used for accessing the parent allocator.
-    */
-    static if (stateSize!Allocator)
+    private template Impl()
     {
-        Allocator _parent;
-        static if (is(Allocator == RCIAllocator))
+        static if (stateSize!Allocator)
         {
-            Allocator parent()
+            Allocator _parent;
+            static if (is(Allocator == RCIAllocator) || is(Allocator == RCISharedAllocator))
             {
-                if (_parent.isNull) _parent = theAllocator;
-                assert(alignment <= _parent.alignment);
-                return _parent;
+                Allocator parent()
+                {
+                    if (_parent.isNull)
+                    {
+                        static if (is(Allocator == RCIAllocator))
+                        {
+                            _parent = theAllocator;
+                        }
+                        else
+                        {
+                            _parent = processAllocator;
```

Comment:
Since parent can be called by multiple threads, is this assignment guaranteed to be thread safe?